server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -449,10 +449,7 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
      * @return convertedValue
      */
     protected Object convertAttributeValue(Object value) {
-        if (value == null || value instanceof String || value instanceof Byte || value instanceof Long || value instanceof Double || value instanceof Boolean || value instanceof Integer || value instanceof Short || value instanceof Float) {
-            return value;
-        }
-        return value.toString();
+        return value;
```

Comment:
what if value is not Serializable ?

Reply:
It is necessaraly serializable, otherwise it can not be passed to setAttributes

Reply:
Contextual information that is not encoded in this code. If it's Serializable, than you can restrict the type of the argument, right ?